### ISSUE TITLE
Quizzes, Tutorials, and other "101" walkthroughs

### ISSUE THREAD

**Issue by OnlineCop** [firasdib/Regex101#604]
A tutorial library might be useful.
- Tutorials would be interactive and explain a single concept at a time.
- Users can move through tutorials in whatever order they want.
- Registered users can upvote/downvote the tutorials they found useful, although votes don't affect the _order_ that tutorials are sorted.
- Tutorials can be maintained in a separate Github repository. It could even be made public so anyone could fork and make PRs, and any bugs or "highly-downvoted" tutorials could be earmarked for review.
- Like the rest of the site, tutorials could be run "offline" once the page is initially loaded.
- Most, if not all, of the [original quiz questions](https://github.com/firasdib/Freenode-regex-quiz/blob/master/quiz.txt) could be implemented.

**Comment by Ouims** [comment]
I think this is a good idea, but at the same time regular-expressions.info is already very good.
Or do you have different kind of tutorials idea in mind? like, how to use some features, for example like how recursive regex are useful to match balanced construct, or (*SKIP)(*FAIL) to skip things you don't want and focus on what you want etc

**Comment by TWiStErRob** [comment]
I think the difference here is that www.regular-expressions.info is a static website talking about the stuff that you can do with regex in various programming languages, but has no facilities to actually walk the walk.

regex101 has the benefit of having the tutorial **and** the try-out area in one place, similar to w3schools, but better. REs are confusing enough by themselvs, there's no need to complicate it further by having to figure out how to embed a regex in a specific language (e.g. Java needs double-escaping, `sed` needs the right arguments and quote type).

This means that here, a student can do a "what happens if I do this" action and experiment freely without having to recompile/run and deal with exceptions and stuff. Granted regex101 can be used as a complimentary service to r-e.info in a separate window, but you would still have to come up with the correct input and way of entering for example the `g` flag.

In summary:  r-e.info explains everything in prose, but there are visual learners like myself. I can learn much faster if I see
![image](https://cloud.githubusercontent.com/assets/2906988/18829368/37bf794c-83dc-11e6-80c0-c3670ef6d9c9.png)
than reading the usual boring "the `+` operator matches one or more occurrences of the repeated term". I could also go in and add more `b`s and observe that it still matches.

**Comment by firasdib** [comment]
The quiz is being brought back, and I'm trying to incorporate regular-expressions.info into it as much as possible. Beyond that, I'm not sure how much I can do here.
